### ISSUE TITLE
[Feature] Allow to only record har files if certain conditions are met and update only if not present.

### ISSUE THREAD

**Issue by chwoerz** [microsoft/playwright#28825]
Tl;dr: [I have created a PR for the following problem](https://github.com/microsoft/playwright/pull/28796)

Currently `routeFromHAR` does always record the results of a test run if specified with `update: true`. This faces the problem that in case of a failed test, non finished files are recorded and the next run if update is disabled, will take the unfinished files and fail for the ones not found or even worse return wrong responses like a 401 when it was recorded. 
I suggest to add an `saveHarFilesOn` property which can then take a function to check if for example a test passed. Like that `saveHARFilesOn: () => testInfo.status === 'passed` 

The second problem, lays in the `Update` property. We have to change it manually to be able to enable or disable recording. I often want to record only the fixtures which do not exist because I know certain tests need new recordings and I delete the fixtures before running the
 test. We currently have created our own stub recorder which checks if the HAR-files exist. If so we do not record it. It would be much nicer to have a setting like `Update: ifNotExists` which only updates the fixtures if they not exist at all. 

By combining these two features we can heavily improve our recording flow of HAR files and even enable stuff like "record har files only if a test failed".

**Comment by chwoerz** [comment]
Had somebody already the chance to look into it?

**Comment by chwoerz** [comment]
@pavelfeldman Could you please check it?

**Comment by mxschmitt** [comment]
Leaving this open for prioritisation.

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you disagree with this closure, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.
